src/plone/restapi/services/navigation/get.py

```diff
@@ -140,6 +140,7 @@ def navtree(self):
             "portal_type": {"query": self.settings["displayed_types"]},
             "Language": self.current_language,
             "is_default_page": False,
+            "sort_on": "getObjPositionInParent",
```

Comment:
@valipod thanks for your fix! Do you think it would be possible to add a test to show/proof objects are properly sorted now with this PR?

Reply:
@tisto This is not my fix, as in the linked issue, the solution was provided by @thomasmassmann. I have just created the pull request because the issue had not moved in the last months, and the bug is creating problems.

Reply:
@valipod, thanks for the merge request. Times are crazy right now...